### PR TITLE
Fix export v1 pipeline

### PR DESCRIPTION
**Pull request by medamineziraoui**
## Why ?
**TL;DR** : the pipeline option `include_jobs` have to respect and act accordingly as it s defined, for true it export the jobs with the pipeline, for false it will export only the pipeline.
And when pipeline fails because the jobs doesn't exist then it should list the missing jobs names.

Currently When we export pipeline from v1 platform using `projectsExportV1`. we export v1 jobs automatically from the platform too. because if those jobs doesn't exist then it will throw error, those changes were made after the **client demanded it**. 
Usually we have to set `include_jobs` to `true` to make that happen.
afterwards the client requested to honor the  `include_jobs` option, means if it's set to false then it will not export all the jobs and will not do it automatically for all the ones contained in the exported v1 pipeline version. 
And if we import a pipeline and it fails because the jobs inside the pipeline version doesn't exist in the v2 then it should list the missing jobs names that are not found.

## Links / Ressources
https://github.com/saagie/gradle-saagie-dataops-plugin/issues/259

## Actual status / Investigations 

When exporting a pipeline from the v1 we remove the `include_job`. we need to check if the exported job exist in the target platform. if not we need to list the missing jobs.

## How?

We delete `include_job`to `true` if we use `projectsExportV1` 

We need 
```
String exportArtifactsV1() {
	logger.debug('Starting Export artifacts v1 task')
	configuration.pipeline.include_job = true // <==== delete this line
```

What we currently have in the `ImportPipelineService`
```
def static getJobsNameFromJobList( jobs, JobsFromPipelines ) {
		
		def jobForPipeVersionArray = []
		
		if (jobs && JobsFromPipelines) {
			def jobsByNames = JobsFromPipelines.name
			jobs.each { job ->
				def existingJob = jobsByNames.find { it.equals(job.name)}
				if(existingJob) {
					jobForPipeVersionArray.add(job.id)
				}
			}
		}
		
		return jobForPipeVersionArray
	}
```

We need to change it and add a ArrayList for the missing jobs and throw expection if we didn't find one 

```
def static getJobsNameFromJobList( jobs, JobsFromPipelines ) {
		
		def jobForPipeVersionArray = []
		def jobsNotFound = []
		if (jobs && JobsFromPipelines) {
			def jobsByNames = JobsFromPipelines.name
			jobs.each { job ->
				def existingJob = jobsByNames.find { it.equals(job.name)}
				if(existingJob) {
					jobForPipeVersionArray.add(job.id)
				} else {
                                        jobsNotFound.add(job.name)
                                }
			}
		}
                 if ( jobsNotFound.size() > 0) {
                     logger.error("Some of the jobs contained in the pipeline doesn't existe in the server side");
                      throw new GraldeException("Job list ${jobsNotFound.toString()}");
                  }
		
		return jobForPipeVersionArray;
	}
```

### CODE REVIEW

File: src/main/groovy/io/saagie/plugin/dataops/tasks/service/importTask/ImportPipelineService.groovy [hunk 1]
```diff
@@ -63,18 +67,24 @@ class ImportPipelineService {
...
 				}
 			}
 		}
+		if ( jobsNotFound.size() > 0) {
+			logger.error("Some of the jobs contained in the pipeline version doesn't exist in targeted platform.");
+			throw new GradleException("Missing job names not found on the target platform => : ${jobsNotFound.toString()}");
+		}
 		
-		return jobForPipeVersionArray
+		return jobForPipeVersionArray;
```
Comment (DecampsRenan) [inline src/main/groovy/io/saagie/plugin/dataops/tasks/service/importTask/ImportPipelineService.groovy:87]: no trailing semicolon in groovy ;)

File: src/main/groovy/io/saagie/plugin/dataops/tasks/service/importTask/ImportPipelineService.groovy [hunk 2]
```diff
@@ -63,18 +67,24 @@ class ImportPipelineService {
...
+				if (existingJob && existingJob.id) {
+					jobForPipeVersionArray.add existingJob.id
+				} else {
+					jobsNotFound.add job
 				}
 			}
 		}
+		if ( jobsNotFound.size() > 0) {
+			logger.error("Some of the jobs contained in the pipeline version doesn't exist in targeted platform.");
+			throw new GradleException("Missing job names not found on the target platform => : ${jobsNotFound.toString()}");
```
Comment (DecampsRenan) [inline src/main/groovy/io/saagie/plugin/dataops/tasks/service/importTask/ImportPipelineService.groovy:84]: No semicolons

File: src/test/groovy/io/saagie/plugin/tasks/artifacts/ArtifactsImportTaskTest.groovy [hunk 3]
```diff
@@ -31,6 +31,7 @@ class ArtifactsImportTaskTest extends DataOpsGradleTaskSpecification {
             }
         """
 		
+
```
Comment (DecampsRenan) [inline src/test/groovy/io/saagie/plugin/tasks/artifacts/ArtifactsImportTaskTest.groovy:34]: Unecessary change
